### ISSUE TITLE
Close tray menu cause application quit on macOS with electron v2.0.0-beta.1

### ISSUE THREAD

**Issue by tzh2014** [electron/electron#12124]
When trying out the electron v2.0.0-beta.1, close the tray menu (no matter click an menu item or not), always cause the application quit. It works normally when using v1.8.2.

Os: Mac OS X 10.13.3
Saw these from the problem report: 
Crashed Thread:        0  CrBrowserMain  Dispatch queue: com.apple.main-thread
Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]

**Comment by codebytere** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

Please provide a standalone test that a tester could run to reproduce the issue. This makes fixing issues go more smoothly: not only does it ensure the right thing is tested, but it also prevents triage from bottlenecking on making testers write new tests for each new issue.

I'm setting the more-information-needed label for the above reasons. After you've responded, please @ me in a followup comment.

Thanks in advance! Your help is appreciated.

**Comment by tzh2014** [comment]
@codebytere Hi thank you very much for the reply. It was just happening on an electron application with a normal tray component, nothing special, so I didn't provide any tester. 

And I have just tested the newer beta versions of the 2.0.0. The tray works normally from the v2.0.0-beta.3, so something related were fixed from the beta.2 to the beta.3 I guess.  

In summary, the bug only happened for electron v2.0.0-beta.1 and electron v2.0.0-beta.2 for me.

Thanks for the help and good work on the electron all the time :clap:
